### DISCUSSION TITLE
Sending events to children services/actors from outside interpreted machines

### DISCUSSION THREAD

**Discussion by geekox86** [statelyai/xstate#3322] [Q&A]
Hi once more.

I have an interpreted machine which has a few spawned actors which in turn have some invoked services (machines).

I am trying to reach the invoked services and send them some events.

```js
const app = interpret(machine).start();
const { actor } = machine.state.context;
const { service } = actor.state.children;

service.send({ type: "EVENT" }); // not working
```

Plz, advise me.

**Comment by davidkpiano** [comment]
✏️ Please create a [CodeSandbox (XState template)](https://codesandbox.io/s/xstate-example-template-qcwed) reproduction of this. Thanks!

Also, try `actor.getSnapshot().children` instead?

You also have to ensure that the `actor` exists before trying to read its snapshot, of course.

**Reply by geekox86** [reply]
Hello and thx for passing by.
I tried both methods but no luck.

What is interesting is that when I converted the actors to normal invoked services and context to children, it started to work.

[Here is the sandbox, check the console output](https://codesandbox.io/embed/unruffled-faraday-sz926x?codemirror=1)

**Reply by davidkpiano** [reply]
Another thing you can try is invoking at the top-level instead of spawning:

```js
createMachine({
  invoke: {
    src: ...
  }
})
```

**Reply by geekox86** [reply]
This is exactly what I did and it worked.
Actors do not work though and I am not sure why?
